libs/community/langchain_community/cache.py

```diff
@@ -392,22 +392,26 @@ def _key(self, prompt: str, llm_string: str) -> str:
     def lookup(self, prompt: str, llm_string: str) -> Optional[RETURN_VAL_TYPE]:
         """Look up based on prompt and llm_string."""
         generations = []
-        # Read from a Redis HASH
-        results = self.redis.hgetall(self._key(prompt, llm_string))
-        if results:
-            for _, text in results.items():
-                try:
-                    generations.append(loads(text))
-                except Exception:
-                    logger.warning(
-                        "Retrieving a cache value that could not be deserialized "
-                        "properly. This is likely due to the cache being in an "
-                        "older format. Please recreate your cache to avoid this "
-                        "error."
-                    )
-                    # In a previous life we stored the raw text directly
-                    # in the table, so assume it's in that format.
-                    generations.append(Generation(text=text))
+        try:
+            # Read from a Redis HASH
+            results = self.redis.hgetall(self._key(prompt, llm_string))
+            if results:
+                for _, text in results.items():
+                    try:
+                        generations.append(loads(text))
+                    except Exception:
+                        logger.warning(
+                            "Retrieving a cache value that could not be deserialized "
+                            "properly. This is likely due to the cache being in an "
+                            "older format. Please recreate your cache to avoid this "
+                            "error."
+                        )
+                        # In a previous life we stored the raw text directly
+                        # in the table, so assume it's in that format.
+                        generations.append(Generation(text=text))
+        # Gotta catch'em all
+        except Exception as e:
+            logger.warning(f"Redis lookup failed: {e}")
```

Comment:
Isn't this an `error` level issue rather than a warning?

Reply:
Yes, will change the log level to error